unix_integration/src/tasks_daemon.rs

```diff
@@ -111,10 +133,34 @@ fn create_home_directory(
         return Err("Invalid/Corrupt home directory path - no prefix found".to_string());
     }
 
+    // Get a handle to the SELinux labeling interface
+    let labeler = match selinux {
+        true => match Labeler::new(&[], true) {
+            Ok(v) => Some(v),
+            Err(_) => {
+                return Err("Failed getting handle for SELinux labeling".to_string());
+            }
+        },
+        false => None,
+    };
+
+    // Construct a path for SELinux context lookups
+    let sel_lookup_path_raw = match selinux {
+        true => match get_user_by_uid(info.gid) {
+            Some(v) => format!("{}{}", home_prefix, v.name().to_str().unwrap()),
+            None => {
+                return Err("Failed looking up username by uid for SELinux relabeling".to_string());
+            }
+        },
+        false => String::new(),
+    };
```

Comment:
`hd_path_raw` may be the UUID of the user (default) or spn instead of their username. The SELinux policy will have contexts defined for `/home/<username>` which is why we ask the policy about the context for `/home/<username>`, otherwise the context will always be `user_u:object_r:user_home_dir_t` instead of `staff_u:object_r:user_home_dir_t` for example.

An alternative to this is to have the daemon create equivalence rules in the policy (so that e.g. `/home/<uuid>` is the same as `/home/<username>` for all contexts), but from a policy perspective that means allowing a process to modify the running policy which is generally a bad idea.

Reply:
Ok, that makes sense from the path perspective, thanks.

Please double check the gid/uid bit :)

Reply:
I think the problem here is that if `/home/<spn>` has a different label to `/home/<username>` if we label `/home/<spn>` with the policy of `/home/<username>` while that will work, on the next restorecon or relable of the FS it will break. So wouldn't that imply an issue with the policy that is on the system?

Reply:
> Please double check the gid/uid bit :)

Since Kanidm uses the GID number for the UID as well, this is doing a Unix user lookup by the same GID number. It admittedly looks confusing, so I can add a comment here to clarify.